components/common-go/log/log.go

```diff
@@ -80,6 +86,31 @@ func Init(service, version string, json, verbose bool) {
 	}
 }
 
+type scrubFormatter struct {
+	delegate log.Formatter
+}
+
+// Format implements logrus.Formatter
+func (formatter *scrubFormatter) Format(entry *logrus.Entry) ([]byte, error) {
+	// scrubbing entry.Message is expensive, we skip it for now
+	// since primary entry.Data should have sensitive data
+
+	for key, value := range entry.Data {
```

Comment:
Maybe it should be default behaviour for `Struct` itself, i.e. traverse without reflect using type checks and only call to reflectwalk for structs. The code will be a bit more complex, since we need to switch between 2 modes, but avoiding using reflection should lead to less allocations and a bit faster processing.

Reply:
That makes a lot of sense. Wanna do that in a separate PR or in this one, given that you've already written the code?

Reply:
I tried it is hard to generalize since type checks should be specific. It did not lead to any performance improvement for logging and everything should be reimplemented twice with type checks or reflection.